src/api/routes/v1/get-network.ts

```diff
@@ -0,0 +1,237 @@
+        if (undefinedValues + errors.length === promises.length) {
+          sendReject()
+        }
+        return
+      }
+      resolved = true
+
+      // resolve with the first available value
+      resolve(value)
+    }
+
+    /**
+     * Helper method when a promise is rejected.
+     *
+     * @param error - The error.
+     */
+    function onError(error: Error): void {
+      // skip if already resolved
+      if (resolved) {
+        return
+      }
+
+      // collect error
+      errors = errors.concat(error)
+
+      // reject promise combinator if all promises are failed/undefined
+      if (undefinedValues + errors.length === promises.length) {
+        sendReject()
+      }
+    }
+
+    promises.forEach(async (promise) => promise.then(onFulfill, onError))
+  })
+}
+
+/**
+ * Fetches the crawl data for the node.
+ *
+ * @param host - The host URL to crawl.
+ * @returns The crawl data from the node.
+ */
+async function fetchCrawls(host: string): Promise<Crawl> {
+  const promises: Array<Promise<Crawl | undefined>> = []
+  for (const port of CRAWL_PORTS) {
+    promises.push(crawlNode(host, port))
+  }
+  return any(promises)
+}
+
+/**
+ * Checks whether the UNL of the node is recorded in the networks table. If so,
+ * then it returns the network associated with the UNL.
+ *
+ * @param unl - The UNL of the node.
+ * @returns Whether the UNL of the node has been seen before.
+ */
+async function getNetworkFromUNL(unl: string): Promise<string | undefined> {
+  const result = await query('networks')
+    .select('id')
+    .where('unls', 'like', `%${unl}%`)
+  return result.length > 0 ? result[0].id : undefined
+}
+
+/**
+ * Checks whether the public key of the node is recorded in the crawls table. If
+ * so, then it returns the network associated with the public key.
+ *
+ * @param publicKey - The public key of the node.
+ * @returns The network associated with the public key of the node. Undefined
+ * if the public key has not been seen by the network.
+ */
+async function getNetworkFromPublicKey(
+  publicKey: string,
+): Promise<string | undefined> {
+  const result = await query('crawls')
+    .select('public_key', 'networks')
+    .where('public_key', '=', publicKey)
+  return result.length > 0 ? result[0].networks : undefined
+}
+
+/**
+ * Adds the node (and its corresponding network) to the networks table.
+ *
+ * @param url - The URL endpoint of the node.
+ * @param unl - The UNL of the node.
+ * @returns The ID of the new network.
+ */
+async function addNode(url: string, unl: string | null): Promise<string> {
+  const newNetwork = (Number(maxNetwork) + 1).toString()
+  maxNetwork += 1
+
+  const network: Network = {
+    id: newNetwork,
+    entry: url,
+    port: 51235,
+    unls: unl ? [unl] : [],
+  }
+  await query('networks').insert({
+    ...network,
+    unls: network.unls.join(','),
+  })
+
+  const crawler = new Crawler()
+  void crawler.crawl(network)
```

Comment:
So we kick off a crawl for that network to find new nodes for the newly added Node?

Reply:
Yes. Otherwise, you'll have to wait until the next set of crawls before you have the nodes from the network.